src/internal.c

```diff
@@ -5842,6 +5842,11 @@ static int DoUserAuthRequestPublicKey(WOLFSSH* ssh, WS_UserAuthData* authData,
                 authFailure = 1;
                 ret = WS_SUCCESS;
             }
+            else if (ret == WOLFSSH_USERAUTH_INVALID_USER) {
+                WLOG(WS_LOG_DEBUG, "DUARPK: public key user rejected");
+                authFailure = 1;
+                ret = WS_SUCCESS;
+            }
             else {
                 authFailure = 1;
```

Comment:
@JacobBarthelmeh I see you have fixed the issue [another way](https://github.com/wolfSSL/wolfssh/pull/559/commits/c651fc72e50c518483a5b9ff176ca1b51ab60342), hence this review can be closed.

Reply:
Thanks for the review @falemagn. It was a good suggestion, I just tried to make it more generic and simplify the code there some since ret was almost always getting set to WS_SUCCESS.